readium/shared/src/main/java/org/readium/r2/shared/extensions/String.kt

```diff
@@ -12,30 +12,39 @@ package org.readium.r2.shared.extensions
 import android.net.Uri
 import java.net.URL
 import java.security.MessageDigest
-import java.util.*
-import org.joda.time.DateTime
-import org.joda.time.DateTimeZone
+import java.text.DateFormat
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
 import org.json.JSONException
 import org.json.JSONObject
 import org.readium.r2.shared.InternalReadiumApi
 
 @InternalReadiumApi
-public fun String.iso8601ToDate(): Date? =
-    try {
-        // We assume that a date without a time zone component is in UTC. To handle this properly,
-        // we need to set the default time zone of Joda to UTC, since by default it uses the local
-        // time zone. This ensures that apps see exactly the same dates (e.g. published) no matter
-        // where they are located.
-        // For the same reason, the output Date will be in UTC. Apps should convert it to the local
-        // time zone for display purposes, or keep it as UTC for storage.
-        val defaultTZ = DateTimeZone.getDefault()
-        DateTimeZone.setDefault(DateTimeZone.UTC)
-        val date = DateTime(this).toDateTime(DateTimeZone.UTC).toDate()
-        DateTimeZone.setDefault(defaultTZ)
-        date
-    } catch (e: Exception) {
-        null
+public fun String.iso8601ToDate(): Date? {
+    for (format in iso8601UtcFormats) {
+        try {
+            return format.parse(this)
+        } catch (ignored: ParseException) {
+            // Continue to the next format if parsing fails.
+        }
     }
+    return null
+}
+
+private val iso8601UtcFormats: List<DateFormat> = listOf(
+    "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
+    "yyyy-MM-dd'T'HH:mm:ss'Z'",
+    "yyyy-MM-dd'T'HH:mm:ss",
+    "yyyy-MM-dd'T'HH:mm",
+    "yyyy-MM-dd"
+).map {
```

Comment:
What's the point of this change? A pattern would be easy to miss.

Reply:
Two reasons:
* Joda is loading a file from the file system when parsing a date string, which is to be avoided from the main thread. 
* Getting rid of the Joda dependency.

But I'll explore this alternative instead: https://github.com/Kotlin/kotlinx-datetime